graph_generator/executor.py

```diff
@@ -28,22 +28,61 @@
 
 @dataclass(order=True)
 class Event:
+    """
+    This class represents an event in the event queue. All events are first
+    ordered by the timestamp. To break ties, use the node config as well as
+    the even priority.
+    """
+
+    @dataclass
+    class WatchDogConfig:
```

Comment:
for my education - is the WatchDogConfig necessarily tied to an Event? / therefore it's intentional to make the WatchDogConfig an inner class of Event?

Reply:
This is largely my fault. I'm borrowing config types to differentiate even types but ran into problems as there is no such thing as WatchdogConfig (it's just an int really)